### PR TITLE
Labs 7 and 8 set toggle status

### PR DESCRIPTION
**Pull request by MeliaBurnley**
Sooooo I'm not sure if I should be using `ToggleStatusModifierResponseHandlerSpy` but it was the only `ToggleStatusModifyResponseHandler` where I could get the last request

### CODE REVIEW

File: tests/Framework/EndpointDummy.php [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 class EndpointDummy implements Endpoint
 {
-    public function execute( Request $r )
+    public function execute( Request $request )
```
Comment (vokoblin) [inline tests/Framework/EndpointDummy.php:10]: DocBlock

File: tests/Framework/EndpointDummy.php [hunk 2]
```diff
@@ -4,7 +4,7 @@
...
     {
         // do nothing.
```
Comment (vokoblin) [inline tests/Framework/EndpointDummy.php:12]: remove it if it does nothing ?

Comment (Clearbooks-Tom) [inline tests/Framework/EndpointDummy.php:12]: it would then not implement the interface

File: src/User/UserToggleStatusModifier.php [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+     * @return JsonResponse
+     */
+    public function execute(Request $request)
+    {
+        if($this->requestIsNotValid($request)) {
+            return new JsonResponse("You didn't include all the necessary information", 400);
+        }
+
+        $groupId = $request->get('groupId');
+
```
Comment (Clearbooks-Tom) [inline src/User/UserToggleStatusModifier.php]: heh you don't need so many blank lines!

File: src/User/UserToggleStatusModifier.php [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    }
+
+    /**
+     * @param Request $request
+     * @param null|string $groupId
+     * @return ModifyToggleRequest
+     */
+    private function createLabsRequest(Request $request, $groupId = null)
+    {
+        return new ModifyToggleRequest($request->get('toggleId'), $request->get('newStatus'), $request->get('userId'), $groupId);
```
Comment (Clearbooks-Tom) [inline src/User/UserToggleStatusModifier.php:69]: yeah this will need updating to use jwt when the frontend supports jwt requests

Comment (Clearbooks-Tom) [inline src/User/UserToggleStatusModifier.php:69]: this is okay as a first iteration though

**Comment by MeliaBurnley** [comment]
There isn't a solid `PermissionService`, so this is using `MockPermissionService` in the DI

**Comment by MeliaBurnley** [comment]
I mentioned why in my commit message, there wasn't another way of getting the previous request. Obviously this will be changed in production code, this was just for the moment

**Pull request merged**